src/lmic/lmic.c

```diff
@@ -1889,6 +1889,9 @@ static bit_t buildDataFrame (void) {
     LMIC.frame[OFF_DAT_HDR] = HDR_FTYPE_DAUP | HDR_MAJOR_V1;
     LMIC.frame[OFF_DAT_FCT] = (LMIC.dnConf | LMIC.adrEnabled
                               | (sendAdrAckReq() ? FCT_ADRACKReq : 0)
+#if !defined(DISABLE_PING)
+                              | ((LMIC.opmode & OP_PINGABLE) ? FCT_CLASSB : 0)
```

Comment:
@terrillmoore I think `LMIC.opmode & OP_PINGABLE` is the wrong check, because this is true after `LMIC_setPingable` is called, but this doesn't mean that the periodic read has started (e.g. the device has no beacon received yet). Is `OP_TRACK` the right mask?

Reply:
I guess. The spec is not clear about this at all. Here's the relevant text (starting at line 1147, page 43, spec 1.0.3
 
> The end-device application requests the LoRaWAN layer 1147 to switch to Class B mode. The LoRaWAN layer in the end-device searches for a beacon and returns either a BEACON_LOCKED service primitive to the application if a network beacon was found and locked or a BEACON_NOT_FOUND service primitive. To accelerate the beacon discovery the LoRaWAN layer MAY use the “DeviceTimeReq” MAC command.

> Once in Class B mode, the MAC layer sets to 1 the Class B bit of the FCTRL field of every uplink frame transmitted.

From context, it does appear possible that they mean the device has to be locked to the beacon. So I think it makes sense to change to `OP_TRACK`, as that bit is set exactly when `EV_BEACON_FOUND` is reported.

Reply:
Fixing as part of #208 major update.